api/src/main/java/org/apache/iceberg/actions/RewriteManifests.java

```diff
@@ -44,6 +47,39 @@ public interface RewriteManifests
    */
   RewriteManifests rewriteIf(Predicate<ManifestFile> predicate);
 
+  /**
+   * Rewrite manifests in a given order, based on partition field names
+   *
+   * <p>Supply an optional set of partition field names to sort the rewritten manifests by. Expects
+   * exact transformed column names used for partitioning; not the raw columnnames that partitions
+   * are derived from. E.G. supply 'data_bucket' and not 'data' for a bucket(N, data) partition
+   * definition
+   *
+   * <p>If not set, manifests will be rewritten in the order of the transforms in the table's
+   * current partition spec.
+   *
+   * @param partitionFieldSortOrder a list of partition field names
+   * @return this for method chaining
+   */
+  default RewriteManifests sort(List<String> partitionFieldSortOrder) {
```

Comment:
One thing to double click here with everyone. I think we originally landed the API name as `sort`, but the implementation is using `partitionByRange`. In that case, I feel `clusterBy` is a more suitable name for this API (and also the other one below). What do we think? @zachdisc @RussellSpitzer @nastra

Reply:
I agree and vote for `clusterBy`. Sounds like we have two votes, I'll refactor unless I hear strong opposition :-)